index.js

```diff
@@ -1,25 +1,26 @@
-var _superagent = require('superagent');
 var methods = require('methods');
-var assign = require('object-assign');
 
-var uses = [];
+module.exports = function(superagent) {
+  var uses = [];
 
-var superagent = assign({}, _superagent);
-
-superagent.use = function(fn) {
-  uses.push(fn);
-  return this;
-};
-
-methods.indexOf('del') == -1 && methods.push('del');
-methods.forEach(function(method) {
-  superagent[method] = function() {
-    var request = _superagent[method].apply(superagent, arguments);
-    uses.forEach(function(use) {
-      request = request.use(use);
-    });
-    return request;
+  superagent.use = function(fn) {
+    uses.push(fn);
+    return this;
   };
-});
 
-module.exports = superagent;
+  if(methods.indexOf('del') === -1) {
+    methdos = methods.slice(0).push('del');
```

Comment:
Also, I've tried this, and that's not how 'push' works. array.push('something') will push that element onto the array but [return the new length of the array](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/push), so 'methods'/'methdos' ends up containing a number (27 in my testing). I had more success with:

```
methods = methods.slice(0)
methods.push('del')
```